Add private badge to repos list

tysongach opened the pull request:
This is an alternative approach to https://github.com/houndci/hound/pull/1296.

**Note:** This is just the design implementation. Functionality for the following items is still needed:

- The `ariaDescribedby` attribute in the React component doesn’t yet render anything.
- The `idName ` attribute in the React component doesn’t yet render anything.
- The logic for only rendering this badge when a repo is indeed private is not implemented (i.e. “Private” is showing on every single repo right now).

![screen shot 2017-06-02 at 16 38 04](https://cloud.githubusercontent.com/assets/903327/26743977/ec8b6676-47b1-11e7-8627-9f9c5bc2d05b.png)

tysongach commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Indicate private repos](https://trello.com/c/LR5FzEhd/818-indicate-private-repos)

gylaz commented:
Any concerns that positioning of the `private` element isn't consistent -- based on the length of the repo name? On advantage having private next to the `Activate` button is that it would have the same positioning, but not sure if that's better.

tysongach commented:
@gylaz That’s a great point. Sometimes inconsistent alignment like that can create a really uneven looking design, but in this case since the badge is not very bold and doesn't impose a lot of visual weight, I think it's fine. I more like that we associate the repo name and its private status via proximity (being so close to each other).

Here's a larger view of a bigger list:

![screencapture-localhost-5000-repos-1496437313683](https://cloud.githubusercontent.com/assets/903327/26744833/a3258c24-47b5-11e7-8d84-c739fbf47eec.png)

gylaz commented:
Good points! I sort of liked that `private` was close to the action button too, since it visually helps to confirm that you're activating a repo that you need to pay for. But I think either way is probably fine.

BTW, looks like snapshots need to be updated.

gylaz commented:
@tysongach should we merge this once tests are green?

tysongach commented:
@gylaz So far this PR is design-only. There’s not yet any logic to properly label private repos as private.

gylaz commented on app/assets/javascripts/components/repo.jsx:
```diff
@@ -41,10 +41,14 @@ class Repo extends React.Component {
+
         <div className={classNames(
           "repo-activation-toggle",
           {"repo-activation-toggle--private": showPrivate}
```
Do we still need this class? Maybe it can be removed/cleaned up now, since we're going the label route.

gylaz commented:
@tysongach I've made this functional. I also removed `ariaDescribedby` and `idName` as they don't seem to be supported by React. Should we merge?

tysongach commented:
@gylaz Great! Yeah, merge away! 👍 

> I also removed ariaDescribedby and idName as they don't seem to be supported by React.

I added these as hopefuls, but let’s go ahead and merge for now. In the future, we can circle back and make these work; it will greatly improve accessibility.
